src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -577,7 +577,15 @@ extern "C" int32_t SystemNative_Link(const char* source, const char* linkTarget)
 extern "C" intptr_t SystemNative_MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     intptr_t result;
+#if HAVE_MKSTEMPS
     while (CheckInterrupted(result = mkstemps(pathTemplate, suffixLength)));
+#elif
+    // mkstemps is not available bionic/Android, but mkstemp is
+    (void)suffixLength; // To prevent the compiler from generating errors because suffixLength is unused
+    while (CheckInterrupted(result = mkstemp(pathTemplate)));
```

Comment:
I don't think this works, at least not with the usage we have of it in CoreFx. mkstemp requires that the last six chars be XXXXXX, but the string we're passing in uses XXXXXX in the middle, hence why we have a suffix and use mkstemps, e.g.
https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs#L182